csharp/src/Ice/OutgoingAsync.cs

```diff
@@ -406,6 +456,49 @@ public override bool Exception(Ice.Exception exc)
             {
                 return ExceptionImpl(ex); // No retries, we're done
             }
+            catch (Ice.RemoteException ex)
+            {
+                return ExceptionImpl(ex); // No retries, we're done
+            }
```

Comment:
AFAICT, this can never happen since `IceHandleException` doesn't transform the exception. Instead of duplicating this method, why not take a `System.Exception` parameter?

Reply:
Yes, I think that's correct - IceHandleException just throws the exception as is. Fixing.
Note that all this code will be refactored when we switch OutgoingAsync to use async/await instead of callbacks.